transport-native-epoll/src/main/c/io_netty_channel_unix_FileDescriptor.c

```diff
@@ -38,3 +145,106 @@ JNIEXPORT int JNICALL Java_io_netty_channel_unix_FileDescriptor_open(JNIEnv* env
     }
     return res;
 }
+
+JNIEXPORT jint JNICALL Java_io_netty_channel_unix_FileDescriptor_write(JNIEnv* env, jclass clazz, jint fd, jobject jbuffer, jint pos, jint limit) {
+    void* buffer = (*env)->GetDirectBufferAddress(env, jbuffer);
+    if (buffer == NULL) {
+        netty_unix_errors_throwRuntimeException(env, "failed to get direct buffer address");
+        return -1;
+    }
```

Comment:
I think we could just test if `GetDirectBufferAddress(...)` returns `NULL` in onLoad(...) and if so throw an exception. As otherwise it is supported by the JVM and we are good.

See also:
https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html
https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#GetDirectBufferCapacity